pkg/minikube/cluster/cluster.go

```diff
@@ -646,3 +671,57 @@ func IsMinikubeRunning(api libmachine.API) bool {
 	}
 	return true
 }
+
+// isMachineExists checks if virtual machine does not exist
+// if the virtual machine exists, return true
+func isMachineExists(vmDriver string, s state.State, err error) (bool, error) {
+	switch vmDriver {
+	case driver.HyperKit:
+		if s == state.Stopped || err.Error() == "connection is shut down" {
+			return false, ErrorMachineNotExist
+		}
+		return true, err
+	case driver.HyperV:
+		if s == state.None {
+			return false, ErrorMachineNotExist
+		}
+		return true, err
+	case driver.KVM2:
+		if s == state.None || s == state.Stopped {
+			return false, ErrorMachineNotExist
+		}
+		return true, err
+	case driver.None:
+		if s == state.Stopped {
+			return false, ErrorMachineNotExist
+		}
+		return true, err
+	case driver.Parallels:
+		if err.Error() == "machine does not exist" {
+			return false, ErrorMachineNotExist
+		}
+		return true, err
+	case driver.VirtualBox:
+		if err == virtualbox.ErrMachineNotExist {
+			return false, ErrorMachineNotExist
+		}
+		return true, err
+	case driver.VMware:
+		if s == state.None || s == state.Stopped {
+			return false, ErrorMachineNotExist
+		}
+		return true, err
+	case driver.VMwareFusion:
+		if s == state.Stopped {
+			return false, ErrorMachineNotExist
+		}
+		return true, err
+	case driver.Mock:
```

Comment:
add a case dor kic driver (Docker)

Reply:
Added kic(docker) case

Reply:
Added other case for kic.
I noticed kic return other behavior depending on whether the kic image is pulled or not.

https://github.com/kubernetes/minikube/pull/5730/files#diff-68ced156bfe1892e288da479d66535f2R244